ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -419,6 +419,44 @@ boolean controlCompactionForBlobStore(PartitionId id, boolean enabled) {
     return succeed;
   }
 
+
+  /**
+   * Add a new BlobStore with given {@link ReplicaId}.
+   * @param replica the {@link ReplicaId} of the {@link Store} which would be added.
+   * @return {@code true} if adding store was successful. {@code false} if not.
+   */
+  boolean addBlobStoreForFileCopy(ReplicaId replica) {
+    rwLock.writeLock().lock();
+    boolean succeed = false;
+    try {
+      if (!running) {
+        logger.error("Failed to add {} because disk manager is not running", replica.getPartitionId());
+      } else {
+        // Directory is already created in prefilecopy steps. So no directory cleanup required.
+        BlobStore store = new BlobStore(replica, storeConfig, scheduler, longLivedTaskScheduler, this, diskIOScheduler,
+            diskSpaceAllocator, storeMainMetrics, storeUnderCompactionMetrics, keyFactory, recovery, hardDelete,
+            replicaStatusDelegates, time, accountService, null, indexPersistScheduler);
+        store.start();
```

Comment:
BlobStore.start would allocate first log segment file 0_0_log, which is not needed for FileCopy.

Reply:
BlobStore.start() will init 0_0_log only if no files present in the partition directory. For our case, since filecopy will have sealed files in the partition directory, it will make the last file as active segment, which is the expected behaviour in State Build